src/templates/nulldata.js

```diff
@@ -14,15 +14,23 @@ function check (script) {
 check.toJSON = function () { return 'null data output' }
 
 function encode (data) {
-  typeforce(types.Buffer, data)
+  typeforce(typeforce.oneOf(types.Buffer, types.Array), data)
 
-  return bscript.compile([OPS.OP_RETURN, data])
+  if (!types.Array(data)) {
+    data = [ data ]
+  }
+
+  return bscript.compile([OPS.OP_RETURN].concat(data))
 }
 
 function decode (buffer) {
   typeforce(check, buffer)
 
-  return buffer.slice(2)
+  var chunks = bscript.decompile(buffer)
+
+  chunks.shift()
+
+  return chunks.length === 1 ? chunks[0] : chunks
```

Comment:
@afk11 remember `decode` has an arbitrary return type,  it merely needs to be able to unpack its outputs back to `encode` to be "complete".
```
encode(...) -> Buffer
decode(Buffer) -> ...
```